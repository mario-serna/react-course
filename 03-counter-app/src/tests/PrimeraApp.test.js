import React from 'react';
import { shallow } from 'enzyme';
// import { render } from '@testing-library/react';
import PrimeraApp from '../PrimeraApp';
import '@testing-library/jest-dom';

describe('Pruebas en <PrimeraApp />', () => {
    // test('debe mostrar el mensaje "Test"', () => {
    //     const saludo = 'Test';
    //     const { getByText } = render(<PrimeraApp saludo={saludo} />);
    //     expect(getByText(saludo)).toBeInTheDocument();
    // });

    test('debe mostrar <PrimeraApp /> correctamente', () => {
        const saludo = 'Test';
        const wrapper = shallow(<PrimeraApp saludo={saludo} />);

        expect(wrapper).toMatchSnapshot();
    });

    test('debe mostrar el subtÃ­tulo enviado por props', () => {
        const saludo = 'Test';
        const subtitulo = 'Subtitulo test';

        const wrapper = shallow(
            <PrimeraApp
                saludo={saludo}
                subtitulo={subtitulo}
            />
        );

        const textoParrafo = wrapper.find('p').text();

        expect(subtitulo).toBe(textoParrafo);
    });


});
