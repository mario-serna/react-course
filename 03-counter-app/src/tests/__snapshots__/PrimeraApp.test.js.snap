// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> debe mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Test
  </h1>
  <p>
    Soy un sub
  </p>
</Fragment>
`;
